src/udev/udev-rules.c

```diff
@@ -183,175 +185,203 @@ struct UdevRules {
         Hashmap *known_users;
         Hashmap *known_groups;
         Hashmap *stats_by_path;
-        UdevRuleFile *current_file;
         LIST_HEAD(UdevRuleFile, rule_files);
 };
 
+#define LINE_GET_RULES(line)                                            \
+        ASSERT_PTR(ASSERT_PTR(ASSERT_PTR(line)->rule_file)->rules)
+
 /*** Logging helpers ***/
 
-#define log_rule_full_errno_zerook(device, rules, level, error, fmt, ...) \
+#define log_udev_rule_internal(device, file, line_nr, level, error, fmt, ...) \
         ({                                                              \
-                UdevRules *_r = (rules);                                \
                 int _lv = (level);                                      \
-                UdevRuleFile *_f = _r ? _r->current_file : NULL;        \
-                UdevRuleLine *_l = _f ? _f->current_line : NULL;        \
+                sd_device *_dev = (device);                             \
+                UdevRuleFile *_f = (file);                              \
                 const char *_n = _f ? _f->filename : NULL;              \
                                                                         \
-                rule_file_mark_issue(_f, _lv);                          \
+                if (_dev && _f)                                         \
+                        _f->issues |= (1U << _lv);                      \
```

Comment:
The `_dev` check introduced in the latest edition doesn't look correct to me: this disables setting of `issues` in all logging based on `log_file_full_errno_zerook`, I suppose it breaks udevadm verify.

Reply:
`udevadm verify` does not process any rules, and `_dev` is always NULL on parsing rules.

Maybe, I should split the change into another commit with description.

Reply:
It _does_ break udevadm verify, testsuite-17.11.sh no longer passes.

Reply:
Ah, I understand my failure, the condition must be inverted. Ouch!

Reply:
In particular, all `log_file_*` primitives are affected.